src/pool.rs

```diff
@@ -165,4 +172,38 @@ impl ConnectionPool {
             Err(e) => Err(Error::from(e)),
         }
     }
-}
+
+
+    fn handshake(&self, client: Client, stream: &mut PooledStream) -> Result<()> {
+        let mut options = FindOptions::new();
+        options.limit = Some(1);
+        options.batch_size = Some(1);
+
+        let flags = OpQueryFlags::with_find_options(&options);
+
+        Cursor::query_with_stream(
+            stream,
+            client,
+            String::from("local.$cmd"),
+            flags,
+            doc! {
+                "isMaster": 1i32,
+                "client": {
+                    "driver": {
+                        "name": ::DRIVER_NAME,
+                        "version": env!("CARGO_PKG_VERSION"),
+                    },
+                    "os": {
+                        "type": ::std::env::consts::OS,
```

Comment:
There are a number of additional options fields that could be passed in here, like the version/name of the actual operating system, but given the purpose of this work is just to identify Rust driver usage, I don't think adding the extra complexity beyond something like `Linux (x86_64)` or `Windows (x86_64)` is worth the effort at this point.